src/wasm-binary.h

```diff
@@ -37,7 +37,8 @@ namespace wasm {
 
 enum {
   // the maximum amount of bytes we emit per LEB
-  MaxLEB32Bytes = 5
+  MaxLEB32Bytes = 5,
+  MaxDataSegments = 100 * 1000 // a hard limit wasm VMs impose
```

Comment:
I wonder if this (collection of limits) should go into abi.h or some similar place; i.e. these are not limitations of the binary format, but web-embedding specific. Maybe even a good way to turn them off and on, although that doesn't necessarily have to be in this PR.

Reply:
Yeah, and another limit is the function size issue (#1246 is open for that), so maybe we should collect up these limits in a central place. `abi` doesn't seem entirely right. Maybe `web-limitations`?

Reply:
I added a new enum for both those limitations (still inside wasm-binary header, since I realized this is a binary format limitation only).